Makefile

```diff
@@ -11,40 +15,33 @@ build:
 	go build -o build/_output/onos-cli-docs-gen ./cmd/onos-cli-docs-gen
 	go build -o build/_output/gnmi_cli ./cmd/gnmi_cli
 
+build-tools:=$(shell if [ ! -d "./build/build-tools" ]; then cd build && git clone https://github.com/onosproject/build-tools.git; fi)
+include ./build/build-tools/make/onf-common.mk
+
+mod-update: # @HELP Download the dependencies to the vendor folder
+	go mod tidy
+	go mod vendor
+mod-lint: mod-update # @HELP ensure that the required dependencies are in place
+	# dependencies are vendored, but not committed, go.sum is the only thing we need to check
+	bash -c "diff -u <(echo -n) <(git diff go.sum)"
+
 test: # @HELP run the unit tests and source code validation
-test: build deps license_check linters
+test: mod-lint build license_check_apache linters license
 	go test github.com/onosproject/onos-cli/pkg/...
 	go test github.com/onosproject/onos-cli/cmd/...
 
+docs: # @HELP generate CLI docs
+docs:
+	go run cmd/onos-cli-docs-gen/main.go
+
 jenkins-test:  # @HELP run the unit tests and source code validation producing a junit style report for Jenkins
-jenkins-test: build-tools deps license_check linters
+jenkins-test: mod-lint build license_check_apache linters license
 	TEST_PACKAGES=github.com/onosproject/onos-cli/... ./../build-tools/build/jenkins/make-unit
 
 coverage: # @HELP generate unit test coverage data
 coverage: build deps linters license_check
 	./../build-tools/build/coveralls/coveralls-coverage onos-cli
 
```

Comment:
do you want to add a target for doc generator? doesn't need to be called but for ease of use

Reply:
I've added is separately, not as part of this.